pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -925,6 +941,99 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, ReentrancyG
         }
     }
 
+    // Circuit Breakers
+
+    /**
+     * @notice Return the full circuit breaker state for the given token.
+     * @dev These are the reference values (BPT price and weight complement) computed when the breaker was set,
+     * along with the percentage bounds. It also returns the current BPT price bounds, needed to check whether
+     * the circuit breaker should trip.
+     */
+    function getCircuitBreakerState(IERC20 token)
+        external
+        view
+        returns (
+            uint256 bptPrice,
+            uint256 weightComplement,
+            uint256 lowerBound,
+            uint256 upperBound,
+            uint256 lowerBptPriceBound,
+            uint256 upperBptPriceBound
+        )
+    {
+        bytes32 circuitBreakerState = _circuitBreakerState[token];
+
+        (bptPrice, weightComplement, lowerBound, upperBound) = CircuitBreakerLib.getCircuitBreakerFields(
+            circuitBreakerState
+        );
+
+        (lowerBptPriceBound, upperBptPriceBound) = CircuitBreakerLib.getCurrentCircuitBreakerBounds(
+            circuitBreakerState,
+            _getNormalizedWeight(token).complement()
+        );
```

Comment:
We may want to split this getter in two. Generally when we want to use the breakers we'll know which direction the price is being pushed in, e.g. we only want to check the lower breaker on the ingoing token on a swap.

This means that during weight changes we'll be doing a costly computation of a bound which we don't need on every swap (x2 as this happens for each token).